modules/boost.algorithm/1.83.0.bzl.1/MODULE.bazel

```diff
@@ -0,0 +1,25 @@
+module(
+    name = "boost.algorithm",
+    version = "1.83.0.bzl.1",
```

Comment:
Do all the boost packages have to resolve to a consistent version?

Reply:
Not sure exactly what you mean? Example?

Reply:
For example, is `boost.algorithm` 1.83.0.bzl.1 going to be usable together with `boost.core` 1.85.0.bzl.1? Minimum version selection ensures that all versions will be as least as high as requested, but there is nothing preventing end users from requesting higher versions of only some, but not all, boost packages.

Reply:
Current solution for this is to bump module compatibility version with each boost release. That way modules can work between patches on the same version ie bcr.1 and bcr.5 but won't work across boost 1.83 and boost 1.84. If anyone has a good way to verify full inter-compatibility between boost release versions, then we can keep the compatibility version the same across those applicable versions that have been tested. Thoughs?

Reply:
I would advice against this: the compatibility level doesn't factor into version resolution, it can only cause the final result to be a failure. This means that bumping the level for every new Boost version will effectively force the end user to resolve all versions manually via overrides.

If you use cyclic dependencies between an umbrella module and the individual modules to keep the versions in sync, end users don't have to resolve anything manually.

Reply:
I would advice against this: the compatibility level doesn't factor into version resolution, it can only cause the final result to be a failure. This means that bumping the level for every new Boost version will effectively force the end user to resolve all versions manually via overrides.

If you use cyclic dependencies between an umbrella module and the individual modules to keep the versions in sync, end users don't have to resolve anything manually.

Reply:
Oh my gosh that's a brilliant idea, and would also allow me to reduce a heap of dep statements. Doing this right now, thank you!

Reply:
Just pushed this change (minus 2k lines of code in bazel_dep lines, yay!🥳)

Can you check I've implemented this correctly and it is indeed what you meant? If I understand correctly, this structure will mean that the boost deps will always get resolved upwards to the highest required boost version for all modules? For example, if a user uses "@boost//:core" with boost 1.83 directly, and then also depends on another module from the BCR that depends on boost core  or another boost module which requires core but a newer or older version, they will both be resolved to the highest common version?

Reply:
Yes, that's exactly what it's supposed to do. I haven't used this pattern yet, but it should work.